src/CPUSolver.cpp

```diff
@@ -550,11 +540,13 @@ double CPUSolver::computeResidual(residualType res_type) {
     FP_PRECISION inverse_k_eff = 1.0 / _k_eff;
     FP_PRECISION* nu_sigma_f;
     Material* material;
+    FP_PRECISION* sigma_s;
 
     for (int r=0; r < _num_FSRs; r++) {
```

Comment:
Sure, I just parallelized all of the residual calculations by FSR. It's worth noting that the only *real* fix for lots of energy groups would be some sort of sparse data structure for scattering matrices, but that can be left for another day...